server/Episememe.Application/Features/MediaFiltering/MediaFilter.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Episememe.Application.DataTransfer;
+using Episememe.Domain.Entities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Episememe.Application.Features.MediaFiltering
+{
+    public class MediaFilter
+    {
+        private SearchMediaDto SearchMedia { get; }
+
+        public MediaFilter(SearchMediaDto searchMedia)
+        {
+            SearchMedia = searchMedia;
+        }
+
+        public IEnumerable<MediaInstance> Filter(IQueryable<MediaInstance> mediaInstances)
+        {
+            IEnumerable<MediaInstance> filteredMedia = mediaInstances.ToList();
+            filteredMedia = IncludeTags(ExcludeTags(InTimeRange(filteredMedia)));
```

Comment:
I think it makes more sense to start with `IncludeTags()`. "Include" filter is most likely to narrow down results to just a few, so other filters will have fewer instances to process